src/ff1.rs

```diff
@@ -49,23 +78,50 @@ impl Radix {
             tmp >>= 1;
         }
         Ok(match log_radix {
-            Some(log_radix) => Radix::PowerTwo { radix, log_radix },
-            None => Radix::Any(radix),
+            Some(log_radix) => Radix::PowerTwo {
+                radix,
+                min_len: cmp::max(
+                    (MIN_RADIX_2_NS_LEN + u32::from(log_radix) - 1) / u32::from(log_radix),
+                    MIN_NS_LEN,
+                ),
+                log_radix,
+            },
+            None => {
+                use libm::{ceil, log10};
+                let min_len = ceil(LOG_10_MIN_NS_DOMAIN_SIZE / log10(f64::from(radix))) as u32;
```

Comment:
[This comment only applies to non-power-of-two radices, and so doesn't affect Zcash.]

I don't particularly like this way of doing it; it's too hard to reason about the rounding. I'd prefer that we calculate the domain size exactly and ensure it is >= 1000000.